wire-ios-request-strategy/Sources/Synchronization/Decoding/StoreUpdateEvent.swift

```diff
@@ -95,6 +112,27 @@ public final class StoredUpdateEvent: NSManagedObject {
         return storedEvent
     }
 
+    static private func storedEventExists(for updateEventHash: Int, in context: NSManagedObjectContext) -> Bool {
```

Comment:
You already when through the trouble of adding event db migrations, perhaps we could go the extra mile and also add a uniqueness constraint on the hash? We could give existing stored events a random hash.

Reply:
alright so there was still more changes but I think it's done now :)